tests/engine/src/commonTest/kotlin/WebSocketEngineTest.kt

```diff
@@ -100,6 +99,8 @@ class WebSocketEngineTest {
     val responseBody = webSocketServer.enqueueWebSocket()
     val connection = webSocketEngine.open(webSocketServer.url())
 
+    delay(1000)
```

Comment:
There is a bunch of refactoring in this PR but I believe this is the main fix. There seem to be a race condition in Ktor. I'm waiting for CI to pass and if everything looks good I'll open a youtrack.